doc/computing/parallelism.rst

```diff
@@ -70,40 +74,47 @@ that increasing the number of workers is always a good thing. In some cases
 it can be highly detrimental to performance to run multiple copies of some
 estimators or functions in parallel (see oversubscription below).
 
-OpenMP-based parallelism
-........................
+Lower-level parallelism with OpenMP
+...................................
 
 OpenMP is used to parallelize code written in Cython or C, relying on
-multi-threading exclusively. By default (and unless joblib is trying to
-avoid oversubscription), the implementation will use as many threads as
-possible.
+multi-threading exclusively. By default, the implementations using OpenMP
+will use as many threads as possible, i.e. as many threads as logical cores.
 
-You can control the exact number of threads that are used via the
-``OMP_NUM_THREADS`` environment variable:
+You can control the exact number of threads that are used either:
 
-.. prompt:: bash $
+ - via the ``OMP_NUM_THREADS`` environment variable, for instance when:
+   running a python script:
+
+   .. prompt:: bash $
+
+        OMP_NUM_THREADS=4 python my_script.py
 
-    OMP_NUM_THREADS=4 python my_script.py
+ - or via `threadpoolctl` as explained by `this piece of documentation
+   <https://github.com/joblib/threadpoolctl/#setting-the-maximum-size-of-thread-pools>`_.
 
-Parallel Numpy routines from numerical libraries
-................................................
+Parallel NumPy and SciPy routines from numerical libraries
+..........................................................
 
-Scikit-learn relies heavily on NumPy and SciPy, which internally call
-multi-threaded linear algebra routines implemented in libraries such as MKL,
-OpenBLAS or BLIS.
+scikit-learn relies heavily on NumPy and SciPy, which internally call
+multi-threaded linear algebra routines of BLAS implemented in libraries
+such as MKL, OpenBLAS or BLIS.
 
-The number of threads used by the OpenBLAS, MKL or BLIS libraries can be set
-via the ``MKL_NUM_THREADS``, ``OPENBLAS_NUM_THREADS``, and
-``BLIS_NUM_THREADS`` environment variables.
+You can control the exact number of threads used by BLAS for each library
+using environment variables, namely:
 
-Please note that scikit-learn has no direct control over these
-implementations. Scikit-learn solely relies on Numpy and Scipy.
+  - ``MKL_NUM_THREADS`` sets the number of thread MKL uses,
+  - ``OPENBLAS_NUM_THREADS`` sets the number of threads OpenBLAS uses
+  - ``BLIS_NUM_THREADS`` sets the number of threads BLIS uses
```

Comment:
```suggestion
  - ``BLIS_NUM_THREADS`` sets the number of threads BLIS uses

Note that BLAS & LAPACK implementations can also be impacted by
`OMP_NUM_THREADS`. To check whether this is the case in your environment,
you can inspect how the number of threads effectively used by those libraries
is affected when running the the following command in a bash or zsh terminal
for different values of `OMP_NUM_THREADS`::

    OMP_NUM_THREADS=2 python -m threadpoolctl -i numpy scipy
```

Reply:
I just have edited this suggestion to add the `.. prompt:: bash $` directive.